File: runelite-client/src/main/java/net/unethicalite/api/script/blocking_events/WelcomeScreenEvent.java
package net.unethicalite.api.script.blocking_events;

import net.runelite.api.MenuAction;
import net.runelite.api.widgets.Widget;
import net.unethicalite.api.widgets.Widgets;
import net.unethicalite.client.Static;

public class WelcomeScreenEvent extends BlockingEvent
{
	@Override
	public boolean validate()
	{
		return isWelcomeScreenOpen();
	}

	@Override
	public int loop()
	{
		Widget playButton = getPlayButton();
		if (playButton != null)
		{
			Static.getClient().interact(1, MenuAction.CC_OP.getId(), -1, playButton.getId());
		}

		return -1;
	}

	public static Widget getPlayButton()
	{
		Widget nu = Widgets.get(413, x -> x.hasAction("Play"));
		if (Widgets.isVisible(nu))
		{
			return nu;
		}

		Widget old = Widgets.get(378, x -> x.hasAction("Play"));
		if (Widgets.isVisible(old))
		{
			return old;
		}

		return null;
	}

	public static boolean isWelcomeScreenOpen()
	{
		return Widgets.isVisible(Widgets.get(378, x -> x.hasAction("Play")))
				|| Widgets.isVisible(Widgets.get(413, x -> x.hasAction("Play")));
	}
}

File: runelite-client/src/main/java/dev/unethicalite/api/script/blocking_events/WelcomeScreenEvent.kt
package dev.unethicalite.api.script.blocking_events

import dev.unethicalite.api.game.Game
import dev.unethicalite.api.widgets.Widgets
import net.runelite.api.MenuAction

class WelcomeScreenEvent : BlockingEvent() {
    companion object {
        fun isOpen(): Boolean {
            val play = Widgets.get(378) { it.hasAction("Play") }
            val newPlay = Widgets.get(413) { it.hasAction("Play") }
            return Widgets.isVisible(play) || Widgets.isVisible(newPlay)
        }
    }

    override fun validate(): Boolean {
        return isOpen()
    }

    override fun loop(): Int {
        val play = Widgets.get(378) { it.hasAction("Play") }
        val newPlay = Widgets.get(413) { it.hasAction("Play") }

        if (Widgets.isVisible(play)) {
            Game.getClient().interact(1, MenuAction.CC_OP.id, -1, play.id)
            return -1
        }

        if (Widgets.isVisible(play)) {
            Game.getClient().interact(1, MenuAction.CC_OP.id, -1, newPlay.id)
            return -1
        }

        return -1
    }
}
